webui/src/Buttons/EditButton.tsx

```diff
@@ -372,6 +373,59 @@ interface TabsSectionProps {
 	feedbacks: FeedbackInstance[]
 }
 
+interface TabStepProps{
+	selectedKey: string | false
+	k: string | number
+	i: number
+	name:string | undefined
+	inputState: {showInputEle:boolean}
+	linkClassname: string | undefined
+	socket: CompanionSocketType
+	controlId: string
+
+}
+function TabStep({selectedKey, k, i, name, inputState, linkClassname, socket, controlId }: TabStepProps){
+
+	const renameStep = useCallback(
+		(stepId: string, newName: string) => {
+			socketEmitPromise(socket, 'controls:step:rename', [controlId, stepId, newName]).catch((e) => {
+				console.error('Failed to rename step:', e)
+			})
+		},
+		[socket, controlId]
+	)
+	
+	function onChangeHandler(e: React.ChangeEvent<HTMLInputElement>){
+		setState({showInputEle: state.showInputEle});
+		if(selectedKey){
+			renameStep(selectedKey, e.target.value);
```

Comment:
The use of `selectedKey` seems strange here. Shouldn't it always be renaming `k`, as that is what this component is?
I know that 99% of the time it should only be possible to open the editor when `selectedKey` and `k` are the same value, but it would simplify things to use `k` to confirm the intent